### PR TITLE
05 stable update tests

### LINKED ISSUE #223: tests: Update tests to enable compatibility with Rust validator_worker

### PR DESCRIPTION
**Pull request by samparsky**
Fix #223

### CODE REVIEW

File: test/integration.js [hunk 1]
```diff
@@ -37,7 +36,7 @@ function aggrAndTick() {
 }
 
 tape('submit events and ensure they are accounted for', async function(t) {
-	const evs = genEvents(3).concat(genEvents(2, 'anotherPublisher'))
+	const evs = genEvents(3).concat(genEvents(2, '0x48b6d5748885988cb657a06647c05fc2af1ffacf'))
```
Comment (Ivshti) [inline test/integration.js]: maybe put this in the dummy data constants (`dummyVals`)

File: test/integration.js [hunk 2]
```diff
@@ -270,7 +261,7 @@ async function testRejectState(t, expectedReason, makeNewState) {
 tape('RejectState: wrong signature (InvalidSignature)', async function(t) {
 	await testRejectState(t, 'InvalidSignature', function(newState) {
 		// increase the balance, so we effectively end up with a new state
-		const balances = { ...newState.balances, someoneElse: '1' }
+		const balances = { ...newState.balances, '0x033ed90e0fec3f3ea1c9b005c724d704501e0196': '1' }
```
Comment (Ivshti) [inline test/integration.js]: again, consider putting this in the `dummyVals` but I have an important question here:

how do we make sure that `0x033` does not override something that's already in `newState.balances`? 

also, in the second case, how do we ensure that '0x033' was not previously present in the `newState.balances`
how about generating a random addr in the second case?

File: test/lib.js [hunk 3]
```diff
@@ -48,14 +50,48 @@ function exec(cmd) {
...
+		} -a dummy -i 0xce07cbb7e054514d590a0262c93070d838bfba2e -u http://localhost:8005 -t`
+		followerTick = `RUST_BACKTRACE=1 ${
+			process.env.RUST_VALIDATOR_WORKER
+		} -a dummy -i 0xc91763d7f14ac5c5ddfbcd012e0d2a61ab9bded3 -u http://localhost:8006 -t`
+	}
+
+	return Promise.all([exec(leaderTick), exec(followerTick)])
+}
+
+function nonce() {
```
Comment (Ivshti) [inline test/lib.js]: is there a more elegant way to do this?

File: test/integration.js (hunk 2 above)
Comment (Ivshti) [reply test/integration.js]: I see now that this is the `creator` from dummyVals

I think that you should indeed gen a random addr here, in this test and in the next (`const fakeBalances=`)

File: test/routes.js [hunk 4]
```diff
@@ -142,11 +142,7 @@ tape('POST /channel: should not work with invalid withdrawPeriodStart', async fu
 			...dummyVals.channel.spec,
 			withdrawPeriodStart: new Date('2200-01-01').getTime(),
 			minPerImpression: '1',
-			maxPerImpression: '1',
-			validators: [
```
Comment (Ivshti) [inline test/routes.js]: This seems to be removed because it was already in the `dummyVals.channel.spec`, correct?

Comment (samparsky) [reply test/routes.js]: yes

File: test/lib.js (hunk 3 above)
Comment (samparsky) [reply test/lib.js]: I made the change to use current timestamp as `nonce`

File: test/integration.js (hunk 2 above)
Comment (samparsky) [reply test/integration.js]: Yeah true, I have added random address generation

File: test/lib.js [hunk 5]
```diff
@@ -48,14 +51,41 @@ function exec(cmd) {
...
 function forceTick() {
-	return Promise.all([
-		exec(
-			`./bin/validatorWorker.js --single-tick --adapter=dummy --dummyIdentity=awesomeLeader --sentryUrl=http://localhost:8005`
-		),
-		exec(
-			`./bin/validatorWorker.js --single-tick --adapter=dummy --dummyIdentity=awesomeFollower --sentryUrl=http://localhost:8006`
-		)
-	])
+	let leaderTick = `./bin/validatorWorker.js --single-tick --adapter=dummy --dummyIdentity=0xce07cbb7e054514d590a0262c93070d838bfba2e --sentryUrl=http://localhost:8005`
```
Comment (Ivshti) [inline test/lib.js]: can we avoid hardcoding those here..we have `dummyVals`

File: test/lib.js [hunk 6]
```diff
@@ -48,14 +51,41 @@ function exec(cmd) {
...
+	let followerTick = `./bin/validatorWorker.js --single-tick --adapter=dummy --dummyIdentity=0xc91763d7f14ac5c5ddfbcd012e0d2a61ab9bded3 --sentryUrl=http://localhost:8006`
+	// using rust validator worker
+	if (process.env.RUST_VALIDATOR_WORKER) {
+		leaderTick = `RUST_BACKTRACE=1 ${
+			process.env.RUST_VALIDATOR_WORKER
+		} -a dummy -i 0xce07cbb7e054514d590a0262c93070d838bfba2e -u http://localhost:8005 -t`
```
Comment (Ivshti) [inline test/lib.js]: can we avoid hardcoding those here..we have `dummyVals`

**Pull request merged**